crates/ln-dlc-node/src/util.rs

```diff
@@ -29,3 +33,52 @@ pub fn build_socket_address(ip: IpAddr, port: u16) -> SocketAddress {
 pub fn into_socket_addresses(address: SocketAddr) -> Vec<SocketAddress> {
     vec![build_socket_address(address.ip(), address.port())]
 }
+
+pub fn parse_from_uuid(uuid: Uuid) -> ReferenceId {
+    let hex = uuid.as_simple().to_hex();
+    let bytes = hex.as_bytes();
+
+    debug_assert!(bytes.len() == 32, "length must be exactly 32 bytes");
+
+    let mut array = [0u8; 32];
+    array.copy_from_slice(bytes);
+
+    array
+}
+
+pub fn parse_from_reference_id(reference_id: ReferenceId) -> Result<Uuid> {
+    let reference_id = hex::decode(reference_id)?;
+    let reference_id = reference_id.to_hex();
+    let reference_id = Uuid::parse_str(&reference_id)?;
+
+    Ok(reference_id)
+}
+
+/// Returns the reference id as printed uuid. If None or invalid the default uuid to string will be
+/// returned. default value: 00000000-0000-0000-0000-000000000000
+pub fn stringify_reference_id(reference_id: Option<ReferenceId>) -> String {
```

Comment:
🔧 I think we want to be converting the `ReferenceId` to a `ProtocolId` and then using the `Display` implementation of that type. Strong types are our friends here.